airbyte-integrations/connectors/source-surveymonkey/source_surveymonkey/streams.py

```diff
@@ -107,15 +108,38 @@ def parse_response(self, response: requests.Response, stream_state: Mapping[str,
             yield response_json
 
 
-class IncrementalSurveymonkeyStream(SurveymonkeyStream, ABC):
+class IncrementalSurveymonkeyStream(SurveymonkeyStream, CheckpointMixin, ABC):
     state_checkpoint_interval = 1000
 
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self._state = None
+
     @property
     @abstractmethod
     def cursor_field(self) -> str:
         pass
 
-    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
+    @property
+    def state(self) -> MutableMapping[str, Any]:
+        return self._state
+
+    @state.setter
+    def state(self, value: MutableMapping[str, Any]):
+        self._state = value
+
+    def read_records(
+        self,
+        sync_mode: SyncMode,
+        cursor_field: List[str] = None,
+        stream_slice: Mapping[str, Any] = None,
+        stream_state: Mapping[str, Any] = None,
+    ) -> Iterable[Mapping[str, Any]]:
+        for record in super().read_records(sync_mode, cursor_field, stream_slice, stream_state):
+            self.state = self.get_updated_state(self.state, record)
```

Comment:
Shouldn't it be "_get_updated_state()" ?

Reply:
Yup, my bad! Fixed, I'll run another regression test to confirm everything's still working as intended